docs/DeveloperGuide.md

```diff
@@ -1,38 +1,280 @@
 # Developer Guide
 
+## Table of Contents
+
++ [Acknowledgements](#acknowledgements)
++ [Design](#design)
```

Comment:
Will you have sections discussing the Expense, FutureExpense, and CurrencyLoader components? It might be better to discuss those separately rather than briefly discussing those classes in the ExpenseManager component section?